server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java

```diff
@@ -202,9 +215,10 @@ public Invocations getInvocations() {
          cachesHeath.put(ch.getCacheName(), ch.getStatus());
       }
 
+      RestCacheManager<Object> restCacheManager = invocationHelper.getRestCacheManager();
       // We rely on the fact that getCacheNames doesn't block for embedded - remote it does unfortunately
       return Flowable.fromIterable(cacheManager.getCacheNames())
-            .map(cacheManager::getCache)
+            .map(cacheName -> restCacheManager.getCache(cacheName, request))
```

Comment:
I don't understand why do you need the `RestCacheManager`.
It only decorates the cache with encodings and flags. Both don't affect the methods `getName()` neither `getCacheConfiguration()`

Reply:
because of the subject

Reply:
if you go to the cache resource you will see

Reply:
so, you are using the `RestCacheManager` just to avoid doing
`.map(cache -> subject == null ? cache : cache.withSubject(subject))` 
?

Reply:
well, not only